### PR TITLE
fix: Fix onConnectHandler panics and update config file

### LINKED ISSUE #285: [Device MQTT] ProfilesDir  & DevicesDir  are not set and duplicate device names
ProfilesDir = './res/profiles'
DevicesDir = './res/devices'

Device name in both device files is the same. 
Do we need both? our can just have the AutoEvents commented out in the one file.

### LINKED ISSUE #286: [Device MQTT] onConnectHandlerhander panics
onConnectHandler hander panics because it is referencing `driver.mqttClient` async and it isn't set yet.

Fix by using `client` that is passed in.

### LINKED ISSUE #287: [Device MQTT] Wrong Error message  when publish data to DataTopic
Publishing the following resulting in error that device can't be found, but logges shown the device is created or already exists.

`"Device MQTT-test-device exists, using the existing one"`

`Incoming reading ignored. No DeviceObject found : topic=DataTopic msg={\n  \"name\" : \"MQTT-test-device\",\n  \"cmd\" : \"testmessage\",\n  \"testmessage\" : \"Hello World\"\n}"`

```
{
  "name" : "MQTT-test-device",
  "cmd" : "testmessage",
  "testmessage" : "Hello World"
}
```

### PR DESCRIPTION
**Pull request by weichou1229**
Close #285 
Close #286
Close #287 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-mqtt-go/blob/master/.github/CONTRIBUTING.md

## What is the new behavior?
- onConnectHandler uses client that is passed in
- Set the ProfilesDir and DevicesDir
- Remove redundant predefined device

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/driver/incominglistener.go [hunk 1]
```diff
@@ -8,47 +8,58 @@ package driver
...
+)
+
 func onIncomingDataReceived(client mqtt.Client, message mqtt.Message) {
 	var data map[string]interface{}
 	json.Unmarshal(message.Payload(), &data)
 
-	if !checkDataWithKey(data, "name") || !checkDataWithKey(data, "cmd") {
+	nameVal, ok := data[name]
+	if !ok {
+		driver.Logger.Warnf("[Incoming listener] Incoming reading ignored, reading data `%v` should contain the field `%s` to indicate the device name", data, name)
```
Comment (lenny-intel) [inline internal/driver/incominglistener.go:30]: The should all be errors, not warnings. They need to be logged when user has log level set to only `ERROR`

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Looks good, just need to tweak the log level to be Error on all the messages when reading is Ignored.

File: internal/driver/incominglistener.go (hunk 1 above)
Comment (weichou1229) [reply internal/driver/incominglistener.go]: Modified. Show error logs when the incoming reading is invalid.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by lenny-intel** [comment]
recheck

**Pull request merged**